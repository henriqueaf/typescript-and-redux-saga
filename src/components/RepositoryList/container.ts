import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';

import RepositoryList from '.';
import { ApplicationState } from '../../store';
import * as RepositoriesActions from '../../store/ducks/repositories/actions';

const mapStateToProps = (state: ApplicationState) => ({
  repositories: state.repositories.data,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(RepositoriesActions, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RepositoryList);
