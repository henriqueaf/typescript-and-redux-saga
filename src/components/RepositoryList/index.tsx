import React, { useEffect } from 'react';

import {
  Repository,
  RepositoriesList,
} from '../../store/ducks/repositories/types';
import RepositoryItem from '../RepositoryItem';

type StateProps = RepositoriesList;

interface DispatchProps {
  loadRequest(): void;
  loadSuccess(data: Repository[]): void;
}

export type Props = StateProps & DispatchProps;

const RepositoryList: React.SFC<Props> = ({ repositories, loadRequest }) => {
  useEffect(() => {
    loadRequest();
  }, []);

  return (
    <ul>
      {repositories.map(repository => (
        <RepositoryItem key={repository.id} repository={repository} />
      ))}
    </ul>
  );
};

export default RepositoryList;
