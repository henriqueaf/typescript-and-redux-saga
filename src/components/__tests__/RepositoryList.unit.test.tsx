import React from 'react';
import * as TestRenderer from 'react-test-renderer';

import { Repository } from '../../store/ducks/repositories/types';
import RepositoryList, { Props } from '../RepositoryList';

const mockLoadRequest = jest.fn();
const mockLoadSuccess = jest.fn();

const defaultProps: Props = {
  repositories: [],
  loadRequest: mockLoadRequest,
  loadSuccess: mockLoadSuccess,
};

const mount = (props?: Object) => {
  const mergedProps = {
    ...defaultProps,
    ...props,
  };

  const wrapper = TestRenderer.create(<RepositoryList {...mergedProps} />);

  return {
    wrapper,
    repositoriesList: wrapper.root.findAllByType('li'),
  };
};

describe('RepositoryList', () => {
  it('Renders an empty ul', () => {
    const { wrapper } = mount();

    const repositoryListInstance = wrapper.root;
    expect(repositoryListInstance.findByType('ul')).not.toBeNull();
  });

  it('Renders a list of li inside an ul', () => {
    const repositories: Repository[] = [
      {
        id: 1,
        name: 'testando 1',
      },
      {
        id: 2,
        name: 'testando 2',
      },
    ];

    const { repositoriesList } = mount({ repositories });
    expect(repositoriesList.length).toBe(2);
  });
});
