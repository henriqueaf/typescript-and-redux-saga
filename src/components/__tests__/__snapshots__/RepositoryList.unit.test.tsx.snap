// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryList Matches snapshot 1`] = `
<ul>
  <li>
    testando 1
  </li>
</ul>
`;
